tensorstore/kvstore/s3/s3_resource.h

```diff
@@ -65,6 +61,85 @@ struct S3RequesterPaysResource
   }
 };
 
+
+/// Specifies S3 endpoint url
+struct S3Endpoint : public internal::ContextResourceTraits<S3Endpoint> {
+ public:
+  static constexpr char id[] = "s3_endpoint";
+  struct Spec {
+    std::optional<std::string> endpoint;
+  };
+  using Resource = Spec;
+
+  static Spec Default() { return {}; }
+  static constexpr auto JsonBinder() {
+    namespace jb = tensorstore::internal_json_binding;
+    return jb::Object(
+        jb::Member("endpoint", jb::Projection(&Spec::endpoint)));
+  }
+  static Result<Resource> Create(
+      const Spec& spec, internal::ContextResourceCreationContext context) {
+    return spec;
+  }
+  static Spec GetSpec(const Resource& resource,
+                      const internal::ContextSpecBuilder& builder) {
+    return resource;
+  }
+};
+
+
+/// Specifies S3 path
```

Comment:
Path is likely covered by the kvstore path member, and can be removed.

Reply:
Removed